plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/DebugNativeAppEngineStandardProjectTest.java

```diff
@@ -91,12 +93,23 @@ public void testDebugLaunch() throws Exception {
     assertNotNull(testProject);
     SwtBotTestingUtilities.waitUntilMenuHasItem(
         bot, () -> bot.menu("Run").menu("Debug As"), endsWith("Debug on Server"));
-    SwtBotTestingUtilities.performAndWaitForWindowChange(
-        bot,
-        () -> {
-          bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
-        });
 
+    bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
+    bot.waitUntil(
+        new DefaultCondition() {
+          @Override
+          public boolean test() throws Exception {
+            return bot.shell("Debug On Server").isOpen();
+          }
+
+          @Override
+          public String getFailureMessage() {
+            return "Cannot find a shell with title 'Debug On Server'";
+          }
+        });
```

Comment:
We should extract this condition -- `waitForShell(Matcher<String> nameMatcher)`.  It feels like this should exist already.

Reply:
We had `waitUntilShellIsClosed`, so I created `waitUntilShellIsOpen`. Unfortunately it has to give the exact text string, unless we make it to traverse every shell to find a match. I think that's overkill for us.